### PR TITLE
[#42] Branch newtype

### LINKED ISSUE #42: Abstract git commands over `Branch` type
- [ ] Create `newtype Branch`
- [ ] Rewrite all GitHub functions that have hardcoded `"master"` branch to take `Branch` as an argument
- [ ] Pass `Branch "master"` to every such function

This is the first step towards #39

### PR DESCRIPTION
**Pull request by nixorn**
Resolves #42

### CODE REVIEW

File: src/Life/Github.hs [hunk 1]
```diff
@@ -85,25 +86,27 @@ cloneRepo (Owner owner) = do
...
+checkRemoteSync (Branch br) = do
+    "git" ["fetch", "origin", br]
+    localHash  <- "git" $| ["rev-parse", br]
+    remoteHash <- "git" $| ["rev-parse", "origin/" <> br]
     pure $ localHash == remoteHash
 
 withSynced :: IO a -> IO a
 withSynced action = insideRepo $ do
+    let branchname = "master"
+        branch = Branch branchname
```
Comment (vrom911) [inline src/Life/Github.hs]: I guess this function also should receive `Branch` and synchronise on it

**Comment by ChShersh** [comment]
@nixorn To make proper pull requests you also should do the following:

1. Add number of issue to the title of pull request.
    _Example:_ `[#42] Branch newtype`
2. Write `Resolves #42` in pull request message. This will close issue automatically when PR is merged.
3. Add `[#42]` prefix to your commit messages

File: src/Life/Github.hs [hunk 2]
```diff
@@ -30,8 +30,9 @@ import Life.Configuration (LifeConfiguration (..), lifeConfigMinus, parseRepoLif
 import Life.Message (chooseYesNo, errorMessage, infoMessage, warningMessage)
 import Life.Shell (lifePath, relativeToHome, repoName, ($|))
 
-newtype Owner = Owner { getOwner :: Text } deriving (Show)
-newtype Repo  = Repo  { getRepo  :: Text } deriving (Show)
+newtype Owner  = Owner  { getOwner  :: Text } deriving (Show)
+newtype Repo   = Repo   { getRepo   :: Text } deriving (Show)
+newtype Branch = Branch { getBranch :: Text } deriving (Show)
```
Comment (ChShersh) [inline src/Life/Github.hs]: Oh, it's a good time to refactor existing code according to modern style-guide suggestions. Let's replace `get` prefix in `newtypes` with `un` prefix.

File: src/Life/Github.hs [hunk 3]
```diff
@@ -44,23 +45,23 @@ askToPushka commitMsg = do
...
     if continue
-    then pushka commitMsg
+    then pushka (Branch "master") commitMsg
     else errorMessage "Abort pushing" >> exitFailure
 
 -- | Make a commit and push it.
-pushka :: Text -> IO ()
-pushka commitMsg = do
+pushka :: Branch -> Text -> IO ()
+pushka (Branch br) commitMsg = do
```
Comment (ChShersh) [inline src/Life/Github.hs]: I think it doesn't make much sense to shorten `branch` to `br`. It's okay to me more verbose.

**Comment by nixorn** [comment]
@vrom911 @ChShersh All requested changes applied. 
I'm not sure about version change. [From this document (par. 2)](https://pvp.haskell.org/#version-numbers) I understood there need minor version change.
@vrom911 what should be added into README.md? There is no changes in usability yet.

File: src/Life/Main/Remove.hs [hunk 4]
```diff
@@ -26,7 +26,7 @@ lifeRemove lPath = whatIsLife >>= \case
     OnlyLife _ -> abortCmd "remove" "dotfiles/ directory doesn't exist"
     OnlyRepo _ -> abortCmd "remove" ".life file doesn't exist"
     -- actual life remove process
-    Both _ _ -> withSynced $ do
+    Both _ _ -> withSynced (Branch "master") $ do
```
Comment (vrom911) [inline src/Life/Main/Remove.hs]: I see that `Branch "master"` will be used a lot, so it's nice to have a separate function for that and to import it from `Life.Github` module 🙂

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,10 @@ Change log
 life-sync uses [PVP Versioning][1].
 The change log is available [on GitHub][2].
 
+# 1.0.2
+
+* Branch as newtype.
```
Comment (vrom911) [inline CHANGELOG.md]: Could you please add the issue number with the link in the following format:

```
* [#42](https://github.com/kowainik/life-sync/issues/42):
  Create newtype `Branch`.
```

Comment (nixorn) [reply CHANGELOG.md]: Done.

**Pull request merged**